cmd/prometheus/main.go

```diff
@@ -187,13 +187,13 @@ func main() {
 		Hidden().PlaceHolder("<bytes>").BytesVar(&cfg.tsdb.WALSegmentSize)
 
 	a.Flag("storage.tsdb.retention", "[DEPRECATED] How long to retain samples in storage. This flag has been deprecated, use \"storage.tsdb.retention.time\" instead").
-		Default(defaultRetentionString).SetValue(&oldFlagRetentionDuration)
+		SetValue(&oldFlagRetentionDuration)
 
 	a.Flag("storage.tsdb.retention.time", "How long to retain samples in storage. Overrides \"storage.tsdb.retention\" if this flag is set to anything other than default.").
-		Default(defaultRetentionString).SetValue(&newFlagRetentionDuration)
+		SetValue(&newFlagRetentionDuration)
```

Comment:
"other than default" in the previous line doesn't make sense anymore from the user's perspective as the flag has no default value anymore.

Internally, the value will be the zero value of `time.Duration`, but that has no meaning to the user.

Suggestion:
```
Overrides \"storage.tsdb.retention\" if this flag is set to any non-zero value. If neither this flag nor \"storage.tsdb.retention\" nor "\storage.tsdb.retention.size\" is set, the retention time defaults to 15d.
```

Reply:
updated